alacritty_terminal/src/ansi.rs

```diff
@@ -1442,39 +1444,56 @@ mod tests {
     };
     use crate::index::{Column, Line};
     use crate::term::color::Rgb;
-    use std::io;
+    use std::io::{self, sink};
 
-    /// The /dev/null of `io::Write`
-    struct Void;
+    struct PseudoTermHandler {
+        index: CharsetIndex,
+        charset: StandardCharset,
+        attr: Option<Attr>,
+        identity_reported: bool,
+    }
 
-    impl io::Write for Void {
-        fn write(&mut self, bytes: &[u8]) -> io::Result<usize> {
-            Ok(bytes.len())
+    impl Handler for PseudoTermHandler {
+        fn terminal_attribute(&mut self, attr: Attr) {
+            self.attr = Some(attr);
         }
 
-        fn flush(&mut self) -> io::Result<()> {
-            Ok(())
+        fn configure_charset(&mut self, index: CharsetIndex, charset: StandardCharset) {
+            self.index = index;
+            self.charset = charset;
         }
-    }
 
-    #[derive(Default)]
-    struct AttrHandler {
-        attr: Option<Attr>,
-    }
+        fn set_active_charset(&mut self, index: CharsetIndex) {
+            self.index = index;
+        }
 
-    impl Handler for AttrHandler {
-        fn terminal_attribute(&mut self, attr: Attr) {
-            self.attr = Some(attr);
+        fn identify_terminal<W: io::Write>(&mut self, _: &mut W) {
+            self.identity_reported = true;
+        }
+
+        fn reset_state(&mut self) {
+            *self = Self::default();
         }
```

Comment:
Do you think it makes sense to define this as a method, when all it does is reset itself to the default anyways? I'm a bit torn on that.

If we replace `handler.reset_state()` with `handler = Handler::default()` we wouldn't really lose anything with regards to conciseness and it's 100% clear what exactly is happening, since we've lost a layer of abstraction. On the other hand `reset_state` is very self descriptive and might be easier to understand than reading `handler = Handler::default()` since the name describes the exact intention of the function.

Reply:
That make sense, but I was thinking about it like parse input, reset state, parse more input, reset state. And this method is a part of a Handler trait, so adding it's implementation maked sense to me. So I'm voting for my approach with using trait's declared method.

Reply:
> And this method is a part of a Handler trait

Oh, that's right, I totally missed that. Let's go with this then.